backend/Node/routes/route-helpers/spotify-helpers.ts

```diff
@@ -20,3 +23,92 @@ export const scopes = [
 	'user-follow-read',
 	'user-follow-modify',
 ];
+
+export const createSpotifyWebApi = () => {
+	return new SpotifyWebApi({
+		clientId: process.env.SPOTIFY_CLIENT_ID,
+		clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
+		redirectUri: 'http://localhost:4000/spotify/callback',
+	});
+};
+
+export async function getPopularityForTracks(req: any, res: any) {
+	var spotifyApi = createSpotifyWebApi();
+	try {
+		if (!req.params.access_token)
+			return res.status(400).send('failed to authenticate');
+		if (!req.body.trackIdArray)
+			return res.status(400).send('trackIdArray is missing');
+
+		spotifyApi.setAccessToken(req.params.access_token);
+		console.log(req.body.trackIdArray);
+
+		const data = await spotifyApi.getTracks(req.body.trackIdArray);
+
+		if (data) {
+			var popularityArr = data.body.tracks;
+			popularityArr = popularityArr.map((track: any) => track.popularity);
```

Comment:
Can we just do `data.body.tracks.map` here? That way we can also define the type for `popularityArr` (and use `const` instead of `var`).

Reply:
implemented